File: StudyAsist-Android/app/src/main/java/com/studyasist/data/local/entity/Goal.kt
package com.studyasist.data.local.entity

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "goals")
data class Goal(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val name: String,
    val description: String? = null,
    val examDate: Long,
    val createdAt: Long = System.currentTimeMillis(),
    val isActive: Boolean = true
)

File: StudyAsist-iOS/StudyAsist/Models/Goal.swift
//
//  Goal.swift
//  StudyAsist
//

import Foundation
import SwiftData

@Model
final class Goal {
    @Attribute(.unique) var id: Int64
    var name: String
    var goalDescription: String?
    var examDate: Int64
    var createdAt: Int64
    var isActive: Bool
    
    init(
        id: Int64 = 0,
        name: String,
        description: String? = nil,
        examDate: Int64,
        createdAt: Int64 = Int64(Date().timeIntervalSince1970 * 1000),
        isActive: Bool = true
    ) {
        self.id = id
        self.name = name
        self.goalDescription = description
        self.examDate = examDate
        self.createdAt = createdAt
        self.isActive = isActive
    }
}
